envoyauth/response.go

```diff
@@ -281,3 +299,64 @@ func (result *EvalResult) GetResponseEnvoyHTTPStatus() (*ext_type_v3.HttpStatus,
 
 	return status, nil
 }
+
+func transformToHTTPHeaderFormat(input interface{}, result *http.Header) error {
+
+	takeResponseHeaders := func(headers map[string]interface{}, targetHeaders *http.Header) error {
+		for key, value := range headers {
+			var headerVal string
+			var ok bool
+			if headerVal, ok = value.(string); !ok {
+				return fmt.Errorf("type assertion error")
+			}
+
+			targetHeaders.Add(key, headerVal)
+		}
```

Comment:
Just trying out an inversion here

```suggestion
		for key, value := range headers {
			if headerVal, ok := value.(string); ok {
				targetHeaders.Add(key, headerVal)
				continue
			}
			return fmt.Errorf("type assertion error")
		}
```

🤷 one seems as good as the other. Let's keep it.